File: New Crunch Time/API/geolocation.js
const axios = require('axios');

const BASE_URL = 'http://ip-api.com/json/';

async function getGeolocationByDomain(domain) {
  const response = await axios.get(`${BASE_URL}${domain}?fields=61439`);
  return response.data;
}

async function getGeolocationByCallerIP() {
  const response = await axios.get(`${BASE_URL}?fields=61439`);
  return response.data;
}

// Usage
getGeolocationByDomain('google.com').then(data => console.log(data));
getGeolocationByCallerIP().then(data => console.log(data));

File: New Crunch Time/geolocation/geolocation.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class GeolocationClient {
    private static final String BASE_URL = "http://ip-api.com/json/";

    public String getGeolocationByDomain(String domain) throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + domain + "?fields=61439"))
                .GET()
                .build();
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public String getGeolocationByCallerIP() throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + "?fields=61439"))
                .GET()
                .build();
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public static void main(String[] args) throws Exception {
        GeolocationClient client = new GeolocationClient();
        System.out.println(client.getGeolocationByDomain("google.com"));
        System.out.println(client.getGeolocationByCallerIP());
    }
}
